src/mavlink-router/endpoint.cpp

```diff
@@ -731,8 +735,12 @@ int UdpEndpoint::open(const char *ip, unsigned long port, bool to_bind)
 ssize_t UdpEndpoint::_read_msg(uint8_t *buf, size_t len)
 {
     socklen_t addrlen = sizeof(sockaddr);
-    ssize_t r = ::recvfrom(fd, buf, len, 0,
-                           (struct sockaddr *)&sockaddr, &addrlen);
+    ssize_t r;
+    if (_mode == UdpEndpoint::Broadcast)
+        r = ::recvfrom(fd, buf, len, 0, NULL, NULL);
+    else
+        r = ::recvfrom(fd, buf, len, 0,
```

Comment:
This change is not needed. You can just ignore the values later

Reply:
sockaddr contains a target IP address. We rewrite it on every incoming packet (recvfrom). To ignore this change we need to add an additional sockaddr structure in the endpoint class.

https://github.com/intel/mavlink-router/blob/da8607430c9e336e826d576c15f555c5f0fe033c/src/mavlink-router/endpoint.cpp#L761